src/metabase/upload.clj

```diff
@@ -755,48 +763,58 @@
     ;; Ideally we would do all the filtering in the query, but this would not allow us to leverage mlv2.
     (model-persistence/invalidate! {:card_id [:in model-ids]})))
 
+(defn- internal-allowlist [m]
```

Comment:
Two thoughts:

1. Name is a bit opaque, would rather name it functionally: `translate-type-keywords` or similar.
2. Do we need to translate between namespaces like this? 
 The [light-weight aliasing](https://clojure.atlassian.net/browse/CLJ-2123) added in 1.11 will make it a cinch.

Reply:
On 1. 20b3c0341e21

On 2.

I am aware of aliasing but I do not see how it helps in this instance. Drivers currently use the `metabase.upload` namespace, not `metabase.upload.types` (see `driver/upload-type->database-type`). I took this to mean there os a public API / internal distinction, hence the choice to preserve the distinction - and thus this function.

BTW there is already a similar keyword munging fn, see `database-type` in `upload.clj`.

Am I missing something obvious?

Reply:
Fair play I guess, yeah I guess we should be consistent.

I don't think lightweight aliasing was a think when I made that call, but I guess also it felt weird to refer to "submodule" namespaces from outside the module. Maybe the wind has just changed inside me, but I would rather avoided the bidirectional translation, feels the greater of two evils.

Curious about your thoughts, but let's punt on it for now.

Reply:
Oh yes, totally agree FWIW. We can do a refactor at some point perhaps to have `types.cljc` use the `:metabase.upload` keys. Also agreed now lightweight aliasing exists this is much easier (but even before I would have been guilty of `(create-ns 'metabase.upload) (alias 'metabase.upload 'upload)`)

Reply:
Well I really don't want to use the top level namespace for the types, then they're indistinguishable from other kinds of namespaced keywords, like the operation type.

There's a part of me that want to go the other way and have `metabase.upload.value-type` and `metabase.upload.column-type` to avoid where we pun on these keywords, but doing the translations between those two is a bit tedious.